runtime/src/main/scala/akka/grpc/internal/ChannelUtils.scala

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.grpc.internal
+
+import java.util.concurrent.CompletionStage
+
+import akka.Done
+import akka.annotation.InternalApi
+import io.grpc.{ ConnectivityState, ManagedChannel }
+
+import scala.compat.java8.FutureConverters._
+import scala.concurrent.{ ExecutionContext, Future, Promise }
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+object ChannelUtils {
+
+  def createDonePromise(): Promise[Done] = Promise[Done]()
```

Comment:
Best method ever!

Reply:
Is that because Promise.apply isn't accessible from Java with Scala 2.11? Perhaps add a comment

Reply:
It is but via the $MONDULE$ and i didn't really want that in the generated code that ppl may look at and we have to maintain without IDE support :)

Reply:
Removed this by having the promise inside a channel wrapper so the clients create it